json_tokener.c

```diff
@@ -298,13 +304,8 @@ struct json_object *json_tokener_parse_ex(struct json_tokener *tok, const char *
 		uselocale(newloc);
 	}
 #elif defined(HAVE_SETLOCALE)
-	{
-		char *tmplocale;
-		tmplocale = setlocale(LC_NUMERIC, NULL);
-		if (tmplocale)
-			oldlocale = strdup(tmplocale);
-		setlocale(LC_NUMERIC, "C");
-	}
+	oldlocale = setlocale(LC_NUMERIC, NULL);
```

Comment:
I'm pretty sure this doesn't actually work w/o making a copy.  The strdup is needed to avoid using memory that is no longer valid after the subsequent setlocale() calls.

Reply:
Yes, you are right. Wow, there are quite some programs out there which do it wrong then. Time to fix these programs and maybe recommend a re-phrasing of the setlocale manual page to clarify this. Obviously removed the commit from my pull request.